### PR TITLE
Task 2: Serving SPA: Completed Manual Deployment

### PR DESCRIPTION
**Pull request by december-man**
All the links are attached to the README file.
Minor changes (2.1.4) were done to the Header.tsx file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,3 +36,23 @@ Runs tests in console, in browser or with coverage.
...
 Runs linting and formatting for all files in `src` folder.
+
+
+
+
+## Manual deployment links
+
+### Cloudfront:
+
+https://d2qyic7cxelodb.cloudfront.net/
```
Comment (takegu) [inline README.md]: Unfortunately, the URL that you provided doesn't work. `403 Forbidden`  
image

Comment (december-man) [reply README.md]: @takegu . This is really weird. I've been playing around with scripts for a while now, while cross-checking. Let me check.

Comment (takegu) [reply README.md]: Ok, it isn't the problem. I'm gonna back later to check your task. Please, let me know when you'll finish

Comment (december-man) [reply README.md]: > Ok, it isn't the problem. I'm gonna back later to check your task. Please, let me know when you'll finish

Thank you! Should be fixed.

Comment (takegu) [reply README.md]: Unfortunately, something went wrong, again :( 
image

Comment (takegu) [reply README.md]: Could you double-check, please?

Comment (december-man) [reply README.md]: > Could you double-check, please?

Most likely your browser cached the page. Everything should work now (Checked from two different devices with different IPs)

Comment (takegu) [reply README.md]: IMG_B38EC870601F-1

Comment (takegu) [reply README.md]: Could you please send me a screenshot of this page?

Comment (takegu) [reply README.md]: Screenshot from different device and the different country

image

Comment (takegu) [reply README.md]: Maybe you provided the wrong URL?

Comment (december-man) [reply README.md]: > Could you please send me a screenshot of this page?

изображение

Comment (december-man) [reply README.md]: Tried using VPN too, it works.

Comment (takegu) [reply README.md]: Oooh, it's some kind of black magic :) But it's ok, *** happens 

Thank you, I'll submit your review in 5 minutes 

Kind regards,
A

Comment (december-man) [reply README.md]: Now I'm really interested. Hold on please. Let's figure this out. The issue must be in the bucket policy and bucket access restriction. How did you restrict access to the bucket: "S3-website shows 403 Access Denied error"? Maybe I've done it incorrectly and it somehow influences the cloudfront data? What I've done, after you've noticed that the access is restricted, is created another invalidation to re-cache the cloudfront data.

Comment (takegu) [reply README.md]: For manual deployment, I used this video as an example [did you see that?](https://videoportal.epam.com/video/lNdwY9xWNBgG96kbay2G)

Comment (december-man) [reply README.md]: > For manual deployment, I used this video as an example [did you see that?](https://videoportal.epam.com/video/lNdwY9xWNBgG96kbay2G)


Yeah, I've seen it. This video doesn't have any information on how to later restrict access directly to S3 bucket. And the distribution was created based off of S3 bucket URL, not the bucket website endpoint.
Meanwhile, in the task it says:
"Check if the app is available through the Internet over http://{your-bucket-name}.s3-website-{aws-region}.amazonaws.com ." (Notice 's3-website-')

Comment (december-man) [reply README.md]: > For manual deployment, I used this video as an example [did you see that?](https://videoportal.epam.com/video/lNdwY9xWNBgG96kbay2G)

https://dy7dvx16hh7jr.cloudfront.net/

Check this out. Just deployed it in 5 minutes. This cloudfront distribution refers to another bucket, and it was created using this origin access control option:
изображение

Comment (takegu) [reply README.md]: If I understand you correctly, 
image
image

Comment (takegu) [reply README.md]: > https://dy7dvx16hh7jr.cloudfront.net/

Now, It works for me

Comment (december-man) [reply README.md]: Yes, thank you. I see the cloudfront OAI. That's what I was missing in my previous solution. Maybe this is the reason behind the weird behavior

Comment (december-man) [reply README.md:47]: > > https://dy7dvx16hh7jr.cloudfront.net/
> 
> Now, It works for me

This bucket's policy:
изображение

Comment (takegu) [reply README.md]: Thank you too. Good luck!

Comment (december-man) [reply README.md]: > Thank you too. Good luck!

You too. Thank you for hanging around!

**Pull request merged**